### PR TITLE
Return timing info in Lean response

### PR DESCRIPTION
**Pull request by khieta**
*Issue #, if available:*

This PR adds timing info to the Lean responses for authorization, validation, and evaluation. Times are obtained using [`IO.monoNanosNow`](https://leanprover-community.github.io/mathlib4_docs/Init/System/IO.html#IO.monoNanosNow). The returned times are printed to the DRT log using `info!`, which matches how we report timing info for the Rust & Dafny.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-drt/src/lean_impl.rs [hunk 1]
```diff
@@ -52,40 +53,40 @@ extern "C" {
     fn initialize_DiffTest_Main(builtin: i8, ob: *mut lean_object) -> *mut lean_object;
 }
 
-#[derive(Debug, Deserialize)]
-struct ListDef<String> {
-    l: Vec<String>,
-}
+pub const LEAN_AUTH_MSG: &str = "Lean authorization time (ns) : ";
```
Comment (andrewmwells-amazon) [inline cedar-drt/src/lean_impl.rs:56]: [nit] personally I'd prefer milliseconds.

File: cedar-lean/Cli/Main.lean [hunk 2]
```diff
@@ -30,7 +30,7 @@ def readFile (filename : String) : IO String :=
 def printUsage (err : String) : IO Unit :=
   IO.println s!"{err}\nUsage: Cli <command> <file>"
 
-def main (args : List String) : IO Unit :=
+unsafe def main (args : List String) : IO Unit :=
```
Comment (andrewmwells-amazon) [inline cedar-lean/Cli/Main.lean:33]: Just out of curiousity, why is this needed now?

**Review by andrewmwells-amazon — APPROVED** [review]

File: cedar-drt/src/lean_impl.rs (hunk 1 above)
Comment (khieta) [reply cedar-drt/src/lean_impl.rs:56]: I don't feel strongly either way, but I like consistency, and the rest of the code uses ns. Do you think it's worth changing the Rust & Dafny timings to use ms too?

File: cedar-lean/Cli/Main.lean (hunk 2 above)
Comment (khieta) [reply cedar-lean/Cli/Main.lean:33]: It's the result of calling `unsafeBaseIO` in `Cli/Main.lean` in order to pull out the timing info from an IO monad. A little hacky -- open to better suggestions 🙂

**Review by emina — APPROVED** [review]
